### DISCUSSION TITLE
Touchlink with Sonoff new USB E dongle EFR32MG21 with Hue

### DISCUSSION THREAD

**Discussion by Loic691** [Koenkk/zigbee2mqtt#13893] [Devices]
Hi
I have an old instance of Z2M with Conbee 2 wich works perfectly.
I installed a fresh VM with Sonoff USB E version new dongle based on EFR32MG21 Silicon labs. I hesitated with old one (P)based on Texas Instrument.

The new Sonoff instance is working with last version of Z2M. I linked several devices without problem.
Note I didn't used a backup from Conbee2 instance to restore on Sonoff Instance. All is new on the Sonoff instance.
I think we can't move these configuration because of different dongle, different IEEE network address, different PAN ID,... Exact ?
The channel is 11 on the two instance (mandatory because I have Legrand device)

I'd like to move two hue Lily and Cala device from old instance to new one but I don't have the philips switch(neither Hue gateway).
So I tryed xthe touchlink seams works OK in the Sonoff EFR32MG21 instance but nothing is detected.
Wich is the exactly procedure ? The Sonoff dongle was at 10 cm from Lily when I tried without success.

Thanks for help

**Comment by kirovilya** [comment]
Unfortunately, the current firmware of Sonoff Dongle-E does not support touchlink.
The backup/restore functions have not yet been implemented for the EZSP adapter in Zigbee2mqtt.

**Reply by Loic691** [reply]
Are you sur because I saw several links wich says référence to touchlink. 
I saw too à docs on gît Thatcher feature is implemented

**Reply by kirovilya** [reply]
can you show links where it is indicated that it works? I checked my stick - the touchlink does not work there

**Reply by Loic691** [reply]
See the below link which is the library EZsP integration in Z2M
https://github.com/Koenkk/zigbee-herdsman/issues/319
Touchlink seams OK
And one of several siLabs doc firmware refer to touchlink
https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.10.3.0.pdf

**Reply by kirovilya** [reply]
:) I know that in other dongles on efr32, the touchlink functionality works. But in the firmware for sonoff dongle-e it is disabled (according to my information). To make it work - you need another firmware (I don’t know which one yet).

**Reply by Hedda** [reply]
> Unfortunately, the current firmware of Sonoff Dongle-E does not support touchlink.

@Daniel-zhan-itead Can you please check if ITead can build new firmware for ZBDongle-E with "touchlink" (touchlinking) support?

https://www.zigbee2mqtt.io/guide/usage/touchlink.html

Question about ezsp adapter touchlink for Zigbee Light Link (ZLL) devices was brought up here -> https://github.com/Koenkk/zigbee-herdsman/issues/319

**Reply by droneman83** [reply]
It say on the first Line
Important: Touchlink only works for adapters based on a Texas Instruments chip (chips starting with CC, e.g. CC2652)
and the E dongle is a EFR32MG21 chip

**Reply by Loic691** [reply]
> It say on the first Line
> Important: Touchlink only works for adapters based on a Texas Instruments chip (chips starting with CC, e.g. CC2652)
> and the E dongle is a EFR32MG21 chip
> 

Yes but this will quickly change as several réponses I had got on differents github projets.
Ezsp is already working for touchlink and is OK in Z2M for some others chips that Cc2652. Docs is not up to date...
For Sonoff EFR32MG21, it's  a firmware items wich be solve in a next release

**Reply by Hedda** [reply]
> It say on the first Line
> Important: Touchlink only works for adapters based on a Texas Instruments chip (chips starting with CC, e.g. CC2652)
> and the E dongle is a EFR32MG21 chip

Yeah the documentation says that now as it is outdated but that is not a software or hardware limitation but rather a firmware build configuration issue which can be solved by EFR32 firmware developers, do Zigbee2MQTT documentation needs to be updated to reflect that EZSP adapters are also supported if Touchlink is enabled in the EFR32 firmware -> https://www.zigbee2mqtt.io/guide/usage/touchlink.html

**Reply by Hedda** [reply]
> Ezsp is already working for touchlink and is OK in Z2M for some others chips that Cc2652. **Docs is not up to date...**


FYI, Touchlink documentation for Zigbee2MQTT has been updated now -> https://github.com/Koenkk/zigbee2mqtt.io/pull/1570

_**Important:** The touchlinking function **only** works with Zigbee Coordinator adapters based on a Texas Instruments ZNP adapters (TI chips starting with "CC", e.g. CC2652) and Silicon Labs EZSP adapters (Silabs chips starting with "EFR32", e.g. EFR32MG21) with touchlink enabled in the Zigbee Coordinator firmware._

https://www.zigbee2mqtt.io/guide/usage/touchlink.html

**Comment by billygr** [comment]
Is there a way to identify through zigbee2mqtt if the touchlink is enabled in the Zigbee Coordinator firmware ?
